viewing_party/party.py

```diff
@@ -1,23 +1,232 @@
+            fantasy_count+=1
+            genre_dict['Fantasy']= fantasy_count
+        elif movie['genre']=='Action':
+            action_count+=1
+            genre_dict['Action']=action_count
+        elif movie['genre']=='Intrigue':
+            intrigue_count+=1
+            genre_dict['Intrigue']=intrigue_count
+        elif movie['genre']=='Horror':
+            horror_count+=1
+            genre_dict['Horror']= horror_count
+
+    #find max_count=max(genre_dict.values())
+    if len(genre_dict)==0:
+        return None
+    else:
+        popular_genre=max(genre_dict, key=genre_dict.get)
+        return popular_genre
+    
+    
+    
+        
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+
+    #create total list of movies
+    list_of_watched_movies=[]
+    for friend in user_data['friends']:
+        for movie in friend['watched']:
+            list_of_watched_movies.append(movie)
+
+    list_of_unique_movies=[]
+    #create list of unique movies   
+    for watched_movie in user_data['watched']:
+        if watched_movie not in list_of_watched_movies:
+            list_of_unique_movies.append(watched_movie)
+
+    return list_of_unique_movies  
+#write helper function
+
+
+def get_friends_unique_watched(user_data):
+
+    #create total list of all watched_movies
+    list_of_watched_movies=[]
+    for friend in user_data['friends']:
+        for movie in friend['watched']:
+            list_of_watched_movies.append(movie)
+
+    
+    #create list of unique movies for user
+    list_of_unique_movies=[]
+    for watched_movie in user_data['watched']:
+        if watched_movie not in list_of_watched_movies:
+            list_of_unique_movies.append(watched_movie)
+
+    #from list_of_watched_movies subtract list_of_unique_movies 
+    friends_unique_movies=[]
+    for unique_movie in list_of_watched_movies:
+        if unique_movie not in user_data['watched']:
+            friends_unique_movies.append(unique_movie)
+
+    #ensure no duplicates in friends_unique_movie_list
+    result=[]
+    for i in friends_unique_movies:
+        if i not in result:
+            result.append(i)
+        print(len(result))
+    friends_unique_movies=result
+    
+    return friends_unique_movies
+    
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+
+    #create list of all movies like wave 3 (movies_to_recommend)
+    movies_to_recommend=[]
+    for friend in user_data['friends']:
+        for movie in friend['watched']:
+            movies_to_recommend.append(movie)
+
+    #create list of all hosts that are available (all_available_hosts)
+    all_available_hosts=[]
+    for host in user_data["subscriptions"]:
+            all_available_hosts.append(host)

```

Comment:
Once again, you're creating a list (all_available_hosts) that already exists (user_data["subscriptions"]). That would make sense if we were modifying the list in some way, but since we are just walking through the list later, the extra list becomes redundant. You could also simply do the following and you would achieve the same result without the for loop:

     all_available_hosts = user_data["subscriptions"]